test/torch/nn/test_rnn.py

```diff
@@ -0,0 +1,182 @@
+    )
+    model = (
+        GRU(
+            input_size=input_size,
+            hidden_size=hidden_size,
+            num_layers=num_layers,
+            batch_first=True,
+            bidirectional=bidirectional,
+        )
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+
+    out, h = model(x)
+    assert out.get().float_precision().shape == th.Size(
+        [batch_size, seq_len, directions * hidden_size]
+    )
+    assert h.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+
+    h0 = (
+        th.randn(batch_size, directions * num_layers, hidden_size)
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+    out, h = model(x, h0)
+    assert out.get().float_precision().shape == th.Size(
+        [batch_size, seq_len, directions * hidden_size]
+    )
+    assert h.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+
+
+def test_lstm(hook, workers):
+    bob = workers["bob"]
+    alice = workers["alice"]
+    crypto_prov = sy.VirtualWorker(hook, id="crypto_prov")
+
+    # model hyperparameters
+    batch_size = 4
+    input_size = 3
+    hidden_size = 5
+    seq_len = 6
+    num_layers = 2
+    bidirectional = True
+    directions = 2 if bidirectional else 1
+
+    x = (
+        th.randn(batch_size, seq_len, input_size)
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+    model = (
+        LSTM(
+            input_size=input_size,
+            hidden_size=hidden_size,
+            num_layers=num_layers,
+            batch_first=True,
+            bidirectional=bidirectional,
+        )
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+
+    out, (h, c) = model(x)
+    assert out.get().float_precision().shape == th.Size(
+        [batch_size, seq_len, directions * hidden_size]
+    )
+    assert h.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+    assert c.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+
+    h0 = (
+        th.randn(batch_size, directions * num_layers, hidden_size)
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+    c0 = (
+        th.randn(batch_size, directions * num_layers, hidden_size)
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+    out, (h, c) = model(x, (h0, c0))
+    assert out.get().float_precision().shape == th.Size(
+        [batch_size, seq_len, directions * hidden_size]
+    )
+    assert h.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+    assert c.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
```

Comment:
I think a test showing that a RNN trains properly (aka loss decreasing) is a good idea.

Reply:
Agree

Reply:
I just found out the RNNs are not working with AutogradTensors. It will take a bit of time to debug and implement a test for RNN training.

Reply:
@mari-linhares I implemented a test with federated learning instead. I will work on AutogradTensors in another PR

Reply:
great!